packages/glimmer-compiler/lib/template-compiler.ts

```diff
@@ -380,7 +408,11 @@ function assertValidHasBlock({ params }): string {
   }
 }
 
-function assertValidHasBlockParams({ params }): string {
+function assertValidHasBlockParams({ params, hash }): string {
+  if (hash && hash.pairs.length > 0) {
+    throw new Error(`has-block-params does not take any named arguments`);
```

Comment:
Needs to include location information, without it the source of this error will very hard to track down in large apps.